Expand DomConverter.blockElements with missing elements

jacekbogdanski opened the pull request:
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (engine): `DomConverter.blockElements` is missing some of the HTML block element names. Closes #9801. Closes #7863.

maxbarnas commented:
All good, restarted the build job. Will wait for it to finish before merging.

Reinmar commented:
> Other (engine): Changed `DomConverter.blockElements` type to `Set.<String>`. Closes #3920.

This sounds like a breaking change. This is a part of a public API and we recommended adding elements to that array.

Secondly, the ticket that I opened in 2016 was opened when this thing was added in the first place and it made more sense then. Changing this now, without a good reason has a dubious ROI in my opinion. The cost here is in the fact that it's a breaking change.

jacekbogdanski commented:
@Reinmar thanks, I've reverted the type change. I've also changed the merge message into a fix, as it seems to be fixing some cases in `code-block` and `markdown-gfm` plugins.
